coverage/PENDING_REVIEW/intel/0461-COVERAGE-nir-search-helpers-h-254/cov-sum-uniform-vector-components-round.amber

```diff
@@ -0,0 +1,225 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific NIR code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+# #define _int_1 _GLF_uniform_int_values[0]
+# #define _int_0 _GLF_uniform_int_values[1]
+# #define _float_1537_5 _GLF_uniform_float_values[0]
+# #define _float_2_0 _GLF_uniform_float_values[1]
+# #define _float_3_0 _GLF_uniform_float_values[2]
+#
+# precision highp float;
+# precision highp int;
+#
+# // Contents of _GLF_uniform_int_values: [1, 0]
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     int _GLF_uniform_int_values[2];
+# };
+#
+# // Contents of _GLF_uniform_float_values: [1537.5, 2.0, 3.0]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     float _GLF_uniform_float_values[3];
+# };
+#
+# // Contents of resolution: [256.5, 256.5]
+# layout(set = 0, binding = 2) uniform buf2
+# {
+#     vec2 resolution;
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     float f = _float_2_0 * resolution.x + _float_3_0 * round(resolution.x) + resolution.y;
```

Comment:
I am worried that this is implementation-defined. The GLSL docs for "round" say "The fraction 0.5 will round in a direction chosen by the implementation, presumably the direction that is fastest."

This means that 256.5 could round to 256.0 or to 257.0. Here you are relying on rounding to 256.0.

One solution would be to change the if to:

`if (f == _float_1537_5 || f == the_other_possible_value)`

Reply:
I'll just change resolution.x to 256.6 to avoid this. It still hits the coverage point with that.